### ISSUE TITLE
support multiple connections in one query editor

### ISSUE THREAD

**Issue by dengc367** [sqlpad/sqlpad#933]
I have a idea, the query editor can support several multiple connection by setup the connection metadata in comment, like the following screenshot, there are three sqls:
The first which have one `conn` metdata line setup `clickhouse` will be use `clickhouse` connection, 
second one will use gp212 connection, 
and the third sql will use default connection `gp234`.
@rickbergfalk, do you have any other ideas which can support multiple connections in one query editor easily?
![image](https://user-images.githubusercontent.com/1353036/104260967-76969980-54bf-11eb-8d1e-a012f475fa88.png)

**Comment by rickbergfalk** [comment]
@dengc367 would this be for using SQLPad as more of a "notebook" style app like Jupyter or something?

I could potentially see this getting difficult to manage since it kind of messes with the data models of things a bit. The connectionId on the query object doesn't really matter anymore since it is driven by something in the SQL document. Things like query history and batch is only capturing some of the data since a batch could hit multiple connections as opposed to just 1.

Instead of embedding this in a sql doc, what if the UI was changed to something more similar to a notebook style interface? 

Franchise is a nice example. It only supports 1 database connection per notebook I believe. I don't see why each block could have its own connection set. https://franchise.cloud/app/

This would be a significant effort however.

**Comment by dengc367** [comment]
No, I don't need the the `notebook` style like `franchise` app. 
Now the SQLPad supports multi-queries with one connection, I just need that can support multi-queries with multi connections and one default connection. 
Like my screenshot above the thread, `sql1` query will execute with `clickhouse` connection, `sql2` query will execute with `gp212` connection,  and `sql3` query will execute with default `gp234` connection.
